src/hb-set.hh

```diff
@@ -506,16 +506,34 @@ struct hb_set_t
 
   bool is_subset (const hb_set_t *larger_set) const
   {
-    if (get_population () > larger_set->get_population ())
-      return false;
+	if(unlikely(larger_set->page_map.length == 0)) {
+		return false;
+	}
+	if(page_map[page_map.length].major > larger_set->page_map[larger_set->page_map.length].major) {
+		return false;
+	}
 
-    /* TODO Optimize to use pages. */
-    hb_codepoint_t c = INVALID;
-    while (next (&c))
-      if (!larger_set->has (c))
-	return false;
+	uint32_t spi = 0;
+	for(uint32_t lpi = 0; spi < page_map.length && lpi < larger_set->page_map.length; lpi++) {
+		uint32_t spm = page_map[spi].major;
+		uint32_t lpm = larger_set->page_map[lpi].major;
+		if(spm < lpm) {
```

Comment:
similar thing here, the page at spi may be all zeros. So you'll only want to return false if `!sp.is_empty()`

Reply:
Done